src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -1067,7 +1073,7 @@ private ImmutableArray<BoundNode> GuardedAddBoundTreeAndGetBoundNodeFromMap(CSha
 
             if (bound != null)
             {
-                alreadyInTree = _guardedNodeMap.ContainsKey(bound.Syntax);
+                alreadyInTree = _guardedNodeMap.ContainsKey(AsNodeMapKey(bound.Syntax));
```

Comment:
This change doesn't feel right. The bound node is associated with a particular syntax, that syntax should be used to cache the node. If the node is associated with wrong syntax, it would be better to change binding to use the right syntax for the node.